internal/resolution/variable_sources/crdconstraints/crd_constraints.go

```diff
@@ -1,4 +1,4 @@
-package crd_constraints
```

Comment:
I'm a bit on the fence with this one. Lookign for opinions. I can revert and we can disable this check.

While valid and compiles this considered to not be idiomatic in Go.

Same goes for `bundles_and_dependencies` and `required_package`.

See:

* https://go.dev/doc/effective_go#package-names
* https://go.dev/blog/package-names

Reply:
Or even just package `constraints` with types `CRD` and `KubernetesVersion`. Or maybe even more combined, package `olm` with `CRDConstraint`, `KubernetesVersionConstraint`, `BundleVariable`, etc.

Do we need all of the package sprawl?

Reply:
@joelanford @everettraven I'm in in favour of starting with just one `constraints` package and splitting it into more once we see the need and/or once/if we decide to make this stuff exportable.

But I think it should be done outside of this PR (which is already fairly big).

Reply:
> But I think it should be done outside of this PR (which is already fairly big).

Yeah, definitely +1